containers/ddev-webserver/ddev-webserver-base-files/usr/local/bin/xhprof_prepend.php

```diff
@@ -1,4 +1,25 @@
 <?php
-if (extension_loaded('xhprof')) {
+
+$uri = "none";
+if (!empty($_SERVER) && array_key_exists('REQUEST_URI', $_SERVER)) {
+    $uri = $_SERVER['REQUEST_URI'];
+}
+
+// Enable xhprof profiling if we're not on an xhprof page
+if (extension_loaded('xhprof') && strpos($uri, '/xhprof') === false) {
     xhprof_enable();
+    register_shutdown_function('xhprof_completion');
+}
+
+// Write to the xhprof_html output and latest on completion
+function xhprof_completion() {
+    $xhprof_link_dir = "/var/www/xhprof/xhprof_html/latest/";
+
+    $xhprof_data = xhprof_disable();
+    $appNamespace = "ddev";
+    include_once '/var/www/xhprof/xhprof_lib/utils/xhprof_lib.php';
+    include_once '/var/www/xhprof/xhprof_lib/utils/xhprof_runs.php';
+
+    $xhprof_runs = new XHProfRuns_Default();
+    $run_id = $xhprof_runs->save_run($xhprof_data, $appNamespace);
```

Comment:
Maybe we want the ability for a dev user to supply their own script here?

Reply:
I have been thinking about that, and definitely agree. Can go so very far into the weeds. I've also thought about having the script optionally add the link at the bottom. But mucking with the output breaks all APIs and all command-line scripts and lots of web pages (like TYPO3).

Reply:
I think it would be cool to have the shutdown function include a file which was easily editable by the user, perhaps empty by default.